pkg/manifests/nginx.go

```diff
@@ -32,6 +32,31 @@ func (n *NginxIngressConfig) PodLabels() map[string]string {
 	return map[string]string{"app": n.ResourceName}
 }
 
+func (n *NginxIngressConfig) IsManaging(ing *netv1.Ingress) bool {
+	if ing.Spec.IngressClassName == nil {
+		return false
+	}
+
+	if *ing.Spec.IngressClassName != n.IcName {
+		return false
+	}
+
+	return true
+}
+
+// IngressControllerName returns the name of the controller deployment and a boolean indicating if the Ingress consumes that controller
+func (n *NginxIngressConfig) IngressControllerName(ing *netv1.Ingress) (string, bool) {
```

Comment:
I'm wondering if it would be clearer to have a function like `DeploymentName()`? The caller would first call `IsManaging`, then if true call `DeploymentName`. A bit dry-er but more logic on the caller side so I'm not sold either way

Reply:
Made an updated to this to just internally call IsManaging first. I'm not sold either way on this either